pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -54,6 +60,24 @@ func (cd *clusterDeploy) sync(key string, cluster *v3.Cluster) error {
 	original := cluster
 	cluster = original.DeepCopy()
 
+	// create cluster role for this cluster's templates and templateversions
```

Comment:
similar to code in `project_cluster_handler.go`, we should move cluster template related code to its own controller, under template package.